### PR TITLE
(WIP) Add support for OSX build

### PR DESCRIPTION
**Pull request by tobyhinloopen**
Working on making an OSX build work. Making it work the ugly way first, finding out how to do it pretty later.
 
todo: 

- [ ] Don't break existing Windows build (find out how current separation of builds works, it seems one is using eclipse/netbeans generated makefiles)
- [ ] Fix music sometimes not playing (is this an existing issue or an OSX-build related issue?)
- [ ] Fix saving & loading config ini file. OSX application bundles can/should only read data within the application bundle, not write within. Configuration file should be stored in `~/Library/Application Support/MyApp/*`
 
OSX Build (as of 2018-01-04): 
[PanelPop.zip](https://github.com/a544jh/panel-pop/files/1603189/PanelPop.zip)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,3 +29,7 @@ Debug keys:
 Misc:
 * `Ctrl`+`F` - toggle fps diplay
 * `Alt`+`Return` - toggle fullscreen
+
+# Building (OSX)
+
+    brew install sdl2 sdl2_mixir sdl2_ttf boost
```
Comment (tobyhinloopen) [inline README.md:35]: this should be `sdl2_mixer`

File: Config/ConfigHandler.h [hunk 2]
```diff
@@ -17,8 +17,8 @@ class ConfigHandler {
 
 	static ConfigHandler& getInstance();
 
-	static constexpr const char* CONFIG_FILENAME = "panelpop.ini";
-	static constexpr const KeyConfig DEFAULT_KEYS = {
```
Comment (tobyhinloopen) [inline Config/ConfigHandler.h]: My compiler really didn't like these `static constexpr` lines for some reason

File: Config/ConfigHandler.h [hunk 3]
```diff
@@ -17,8 +17,8 @@ class ConfigHandler {
...
+	const char* CONFIG_FILENAME = "~/Library/Application Support/panel-pop/.panelpop";
+	const KeyConfig DEFAULT_KEYS = {
```
Comment (tobyhinloopen) [inline Config/ConfigHandler.h:21]: My compiler really didn't like these `static constexpr` lines for some reason.

**Comment by a544jh** [comment]
Thanks for the first PR!

The Makefiles are currently generated by NetBeans so they shouldn't really be edited, but I have thought of rewriting them completely manually. I'm currently targeting Linux and Windows (MinGw), but it would of course be nice to have support for MacOS as well.

We could have a separate build script for each platform's distribution package. I have been thinking of making a Debian package as well, so the location for the assests and config files would have to be different for that one as well. For now I try to keep the "installation" as simple as possible by having everything in the same directory, but I see MacOS doesn't really allow this.

I may at some point try to simplify the build process to make it IDE independent, but if you're up to work on it as well, I'm happy to accept a PR!

**Comment by tobyhinloopen** [comment]
Closing for now, might retry/continue some time in the future

**Pull request closed without merging**